File: src/test/java/net/slipp/service/tag/TagHelperTest.java
package net.slipp.service.tag;

import static org.hamcrest.CoreMatchers.*;
import static org.junit.Assert.*;

import java.util.Set;

import net.slipp.domain.tag.Tag;

import org.junit.Test;

import com.google.common.collect.Sets;

public class TagHelperTest {

	@Test
	public void denormalizedTags() {
		Set<Tag> tags = Sets.newHashSet(Tag.newTag("java"), Tag.newTag("javascript"));
		String actual = TagHelper.denormalizedTags(tags);
		assertThat(actual, is("java,javascript"));
	}

}

File: src/test/scala/net/slipp/domain/qna/TagHelperTest.scala
package net.slipp.domain.qna

import com.google.common.collect.Sets
import net.slipp.support.test.UnitTest
import org.junit.Assert._
import org.junit.Test

class TagHelperTest extends UnitTest {
  @Test def denormalizedTags {
    val tag1 = aSomeTag(name = "spring")
    val tag2 = aSomeTag(name = "javascript")
    val tags = Sets.newHashSet(tag1, tag2)
    val actual = TagHelper.denormalizedTags(tags)
    assertTrue(actual.contains(tag1.getName))
    assertTrue(actual.contains(tag2.getName))
  }
}